Gems/Multiplayer/Code/Source/Components/SimplePlayerSpawnerComponent.cpp

```diff
@@ -152,13 +152,30 @@ namespace Multiplayer
             GetNetworkEntityManager()->CreateEntitiesImmediate(prefabEntityId, NetEntityRole::Authority, transform);
 
         NetworkEntityHandle controlledEntity;
-        if (!entityList.empty())
+        if (entityList.empty())
         {
+            // Failure: The player prefab has no networked entities in it.
+            AZLOG_ERROR(
+                "Attempt to spawn prefab '%s' failed, no entities were spawned. Ensure that the prefab contains a single entity "
+                "that is network enabled with a Network Binding component.",
+                prefabEntityId.m_prefabName.GetCStr());
+        }
+        else if (entityList.size() == 1)
+        {
+            // Success: The player prefab has exactly one networked entity in it.
             controlledEntity = entityList[0];
         }
         else
         {
-            AZLOG_WARN("Attempt to spawn prefab %s failed. Check that prefab is network enabled.", prefabEntityId.m_prefabName.GetCStr());
+            // Failure: The player prefab has too many networked entities in it.
+            AZLOG_ERROR(
+                "Attempt to spawn prefab '%s' failed, it contains too many networked entities. "
+                "A player prefab must only have a single networked entity inside of it.",
```

Comment:
what happens if you have multiple networked entities?  Does it only work if the player prefab is the first one in the order or does something really bad happen?  reason I ask is I recall an occasion where a game was spawning multiple entities for players that were networked - one that was the visible character with all the various skin attachments, and another that was the character model used for silhouettes for when the player needed to be drawn behind walls (low-res proxy).

Reply:
Right now, the other entities are getting untracked and possibly despawned. I'm not 100% sure, I haven't exactly gone tracing it to see _precisely_ what's happening. This came up though because a user was trying to spawn a multi-entity prefab using the player spawner, and it was breaking pretty badly: https://github.com/o3de/o3de/issues/16020 . 

This code gets back a list of spawned entities that it promptly ignores except for the first one, so in a best-case scenario those entities get immediately despawned. In a worst-case scenario, they leak and _never_ despawn. In either case, the other entities aren't associated with the player entity, they aren't passed around with OnPlayerJoin / OnPlayerLeave, only one will get set to Autonomous and the others won't, only one will get respawned on a Rejoin, etc.

I'm not personally enamored of this design, but until / unless the design changes, we should at least explicitly mark it as an error if someone tries to use this component in a way that isn't supported.